### ISSUE TITLE
scalacheck-identified TopPctCMS failure for Short, Int, success for Long

### ISSUE THREAD

**Issue by sritchie** [twitter/algebird#569]
This transient error showed up in the Travis test run for #556. Here's the failure:

```
import com.twitter.algebird._

val (depth, width, seed) = (3, 24, -5063)
val data: Seq[Short] = Seq(1, 2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 5)
val cms = TopPctCMS.aggregator[Short](depth, width, seed, 0.1).apply(data)

assert(cms.heavyHitters == Set(2, 3, 4, 5)) // fails!
```

This ALSO fails when `data: Seq[Int]`. It PASSES when `data: Seq[Long]`.

**Comment by sritchie** [comment]
Got this again in a recent test run:


```
[info] - should work as an Aggregator when created from a single, small stream *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: Set(5, 1, 2, 3, 4) was not equal to Set(2, 3, 4, 5)
[info]     Location: (CountMinSketchTest.scala:632)
[info]     Occurred when passed generated values (
[info0m]       depth = 2,
[info]       width = 36,
[info]       seed = 208719 // 66 shrinks
[info]     )
```

**Comment by sritchie** [comment]
And again... it's seed dependent. Here's a better example:

```scala
// the actual "heavy hitter" is 5
val data = Seq(1, 2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 5)

def getHeavy(seed: Int) = {
  val depth = 2
  val width = 224
  TopNCMS.aggregator[Int](depth, width, seed, 1)
    .apply(data).heavyHitters
  }

(1000 to 10000).map(getHeavy(_)).toSet
// Set(Set(5), Set(2), Set(4), Set(1))

// Examples of seeds that produce each value:
(1000 to 10000).map(seed => getHeavy(seed) -> seed).toMap
// Map(
//   Set(5) -> 10000,
//  Set(2) -> 1330,
//  Set(4) -> 9275,
//  Set(1) -> 7168)
```

**Comment by johnynek** [comment]
this is a noisy test, we should just fix the `seed` and move on.

**Comment by sritchie** [comment]
Closed by #594
